spec/models/request_spec.rb

```diff
@@ -4,4 +4,25 @@
 
   it { should validate_presence_of(:name) }
   it { should validate_presence_of(:content) }
+
+  describe '#switch_context' do
```

Comment:
@bzwei Not sure what this is testing. This logic has been tested in other places in the common gem for the request headers. Shouldn't your test case be running the Auto Approval test and checking if it ran in the proper context as opposed to testing the switch_context.

Reply:
This is the unit test for the `#switch_context` method. The test validates current request and tenant are the same as what are recorded in the request context.